silencer-plugin/src/main/scala/com/github/ghik/silencer/SilencerPlugin.scala

```diff
@@ -11,21 +11,30 @@ class SilencerPlugin(val global: Global) extends Plugin { plugin =>
   val description = "Scala compiler plugin for warning suppression"
   val components: List[PluginComponent] = List(component)
   private var globalFilters = List.empty[Regex]
+  private var globalPathFilters = List.empty[Regex]
 
   private lazy val reporter =
-    new SuppressingReporter(global.reporter, globalFilters)
+    new SuppressingReporter(global.reporter, globalFilters, globalPathFilters)
 
   override def processOptions(options: List[String], error: String => Unit): Unit = {
     options.foreach { opt =>
       if (opt startsWith "globalFilters=") {
         globalFilters = opt.drop(14).split(";").map(_.r).toList
       }
+
+      if (opt startsWith "globalPathFilters=") {
+        globalPathFilters = opt.drop(18).split(";").map(_.r).toList
```

Comment:
Can we please refactor this and also `globalFilters` handling by doing something like `opt.split("=", 2)` + pattern matching instead of passing literal string lengths?

Reply:
I did think it was weird the way it was done, but I tried to keep it in line with how it was

Reply:
I created a `FilterType` interface that allows to expand the filters in the future if need be

Reply:
Yes, I missed that in the review of the PR that introduced it.